packages/vite/src/node/plugins/importAnalysis.ts

```diff
@@ -753,9 +753,19 @@ export function importAnalysisPlugin(config: ResolvedConfig): Plugin {
       // normalize and rewrite accepted urls
       const normalizedAcceptedUrls = new Set<string>()
       for (const { url, start, end } of acceptedUrls) {
-        const [normalized] = await moduleGraph.resolveUrl(toAbsoluteUrl(url))
+        let normalized
+        const resolved = await this.resolve(url, importerModule.id || undefined)
+        if (resolved?.id) {
+          const mod = moduleGraph.getModuleById(resolved.id)
+          normalized = mod?.url
+        }
+        if (!normalized) {
+          const [resolved] = await moduleGraph.resolveUrl(toAbsoluteUrl(url))
+          normalized = resolved
+        }
         normalizedAcceptedUrls.add(normalized)
-        str().overwrite(start, end, JSON.stringify(normalized), {
+        const hmrAccept = normalizeHmrUrl(normalized)
+        str().overwrite(start, end, JSON.stringify(hmrAccept), {
           contentOnly: true,
         })
       }
```

Comment:
I think `this.entryEntryFromUrl` in `moduleGraph.ts` needs to be changed because we now pass an id instead of a URL to `moduleGraph.updateModuleInfo` in line 804.
https://github.com/vitejs/vite/blob/1379bdbc82f2317c1e0c2797a4dd5c17ff597068/packages/vite/src/node/server/moduleGraph.ts#L273-L276

Reply:
we still pass url

Reply:
My bad, looks good.